python-checks/src/main/java/org/sonar/python/checks/NumpyIsNanCheck.java

```diff
@@ -48,17 +57,63 @@ private static void checkForIsNan(SubscriptionContext ctx) {
   }
 
   private static void checkOperand(SubscriptionContext ctx, Expression operand, BinaryExpression be) {
-    HasSymbol cast;
-    if (operand.is(Tree.Kind.QUALIFIED_EXPR)) {
-      cast = (QualifiedExpression) operand;
-    } else if (operand.is(Tree.Kind.NAME)) {
-      cast = (Name) operand;
+    Optional.ofNullable(getSymbolFromExpression(operand))
+      .map(Symbol::fullyQualifiedName)
+      .filter("numpy.nan"::equals)
+      .ifPresent(fqn -> {
+        PreciseIssue issue = ctx.addIssue(be, MESSAGE);
+        addQuickFix(issue, operand, be);
+      });
+  }
+
+  @CheckForNull
+  private static Symbol getSymbolFromExpression(Expression symbol) {
+    if (symbol.is(Tree.Kind.QUALIFIED_EXPR)) {
+      return ((QualifiedExpression) symbol).symbol();
+    } else if (symbol.is(Tree.Kind.NAME)) {
+      return ((Name) symbol).symbol();
     } else {
-      return;
+      return null;
+    }
+  }
+
+  private static void addQuickFix(PreciseIssue issue, Expression operand, BinaryExpression be) {
+    if (be.leftOperand().equals(operand)) {
+      getQuickFix(issue, operand, be.rightOperand(), be);
+    } else {
+      getQuickFix(issue, operand, be.leftOperand(), be);
     }
-    Symbol symbol = cast.symbol();
-    if (symbol != null && "numpy.nan".equals(symbol.fullyQualifiedName())) {
-      ctx.addIssue(be, "Don't perform an equality check against \"numpy.nan\".");
+  }
+
+  private static void getQuickFix(PreciseIssue issue, Expression nanOperand, Expression otherOperand, BinaryExpression be) {
+    Optional.of(nanOperand)
+      .filter(op -> op.is(Tree.Kind.QUALIFIED_EXPR))
+      .map(QualifiedExpression.class::cast)
```

Comment:
```java
.filter(op -> op.is(Tree.Kind.QUALIFIED_EXPR))
      .map(QualifiedExpression.class::cast)
```
could be just replaced with
```java
.flatMap(TreeUtils.toOptionalInstanceOfMapper(QualifiedExpression.class))
```

Reply:
Simplidied in b480327fad4c01af3a47bedad7ab2ab595588863.